packages/menu-bar/test/keyboard-navigation.test.js

```diff
@@ -0,0 +1,460 @@
+        },
+        { text: 'Menu Item 2' },
+        {
+          text: 'Menu Item 3',
+          children: [{ text: 'Menu Item 3 1' }, { text: 'Menu Item 3 2' }],
+        },
+        { text: 'Menu Item 4', children: [{ text: 'Menu Item 4 1' }] },
+      ];
+      await nextUpdate(menu);
+      buttons = menu._buttons;
+      subMenu = menu._subMenu;
+      overlay = subMenu._overlayElement;
+    });
+
+    describe('default mode', () => {
+      it('should switch menubar button with items and open submenu on item Arrow Left', async () => {
+        buttons[0].focus();
+
+        await sendKeys({ press: 'ArrowDown' });
+        await nextRender();
+        expect(subMenu.opened).to.be.true;
+        expect(document.activeElement).to.equal(overlay.querySelector('vaadin-menu-bar-item'));
+
+        // Move to previous button with submenu
+        await sendKeys({ press: 'ArrowLeft' });
+        await nextRender();
+
+        expect(subMenu.opened).to.be.true;
+        expect(subMenu.listenOn).to.equal(buttons[3]);
+      });
+
+      it('should switch menubar button without items and focus it on item Arrow Left', async () => {
+        buttons[2].focus();
+
+        await sendKeys({ press: 'ArrowDown' });
+        await nextRender();
+        expect(subMenu.opened).to.be.true;
+        expect(document.activeElement).to.equal(overlay.querySelector('vaadin-menu-bar-item'));
+
+        // Move to previous button without submenu
+        await sendKeys({ press: 'ArrowLeft' });
+        await nextRender();
+
+        expect(subMenu.opened).to.be.false;
+        expect(document.activeElement).to.equal(buttons[1]);
+      });
+
+      it('should switch menubar button with items and open submenu on item Arrow Right', async () => {
+        buttons[2].focus();
+
+        await sendKeys({ press: 'ArrowDown' });
+        await nextRender();
+        expect(subMenu.opened).to.be.true;
+        expect(document.activeElement).to.equal(overlay.querySelector('vaadin-menu-bar-item'));
+
+        // Move to next button with submenu
+        await sendKeys({ press: 'ArrowRight' });
+        await nextRender();
+
+        expect(subMenu.opened).to.be.true;
+        expect(subMenu.listenOn).to.equal(buttons[3]);
+      });
+
+      it('should switch menubar button without items and focus it on item Arrow Right', async () => {
+        buttons[0].focus();
+
+        await sendKeys({ press: 'ArrowDown' });
+        await nextRender();
+        expect(subMenu.opened).to.be.true;
+        expect(document.activeElement).to.equal(overlay.querySelector('vaadin-menu-bar-item'));
+
+        // Move to next button without submenu
+        await sendKeys({ press: 'ArrowRight' });
+        await nextRender();
+
+        expect(subMenu.opened).to.be.false;
+        expect(document.activeElement).to.equal(buttons[1]);
+      });
+
+      it('should switch submenu again on subsequent item Arrow Left', async () => {
```

Comment:
This test was added in https://github.com/vaadin/vaadin-menu-bar/pull/24 - back then, the logic was a bit different (it was refactored to `KeyboardDirectionMixin` much later). So maybe it's not needed anymore, however, I decided to keep it and added a similar test for Arrow Right for consistency.